chain/src/txhashset/txhashset.rs

```diff
@@ -1307,12 +1307,12 @@ impl<'a> Extension<'a> {
 		let mut kern_count = 0;
 		let total_kernels = pmmr::n_leaves(self.kernel_pmmr.unpruned_size());
 		let mut tx_kernels: Vec<TxKernel> = Vec::with_capacity(KERNEL_BATCH_SIZE);
-		for n in 1..self.kernel_pmmr.unpruned_size() + 1 {
+		for n in 1..=self.kernel_pmmr.unpruned_size() {
 			if pmmr::is_leaf(n) {
 				let kernel = self
 					.kernel_pmmr
 					.get_data(n)
-					.ok_or::<Error>(ErrorKind::TxKernelNotFound.into())?;
+					.ok_or_else(|| Error::from(ErrorKind::TxKernelNotFound))?;
```

Comment:
Do we actually need `Error::from` here? We have a `From<ErrorKind>` impl on `Error`, so
```
.ok_or(ErrorKind::TxKernelNotFound)?;
```
might suffice